tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -71,8 +71,11 @@ public static void Initialize()
             try
             {
                 Log.Debug("Sending CallTarget integration definitions to native library.");
-                var definitions = InstrumentationDefinitions.GetAllDefinitions();
-                NativeMethods.InitializeProfiler(definitions);
+                var settings = TracerSettings.FromDefaultSources();
+                settings.Freeze();
```

Comment:
I guess there's a potential issue here when users override the default tracer settings programmatically? e.g. environment variables disable the `Kafka` integrations, but then they do something like this:

```csharp
var settings = TracerSettings.FromDefaultSources();
settings.DisabledIntegrationNames.Clear();
Tracer.Instance = new Tracer(settings);
```

Now we _won't_ be tracing Kafka, but they won't understand why. (The other way around, where they _add_ a disabled integration is ok, as we'll still rejit it, but will ignore it at runtime)

This seems like a fundamental issue with programmatic configuration. We can't guarantee _when_ a user is going to do stuff like this, so we can't ever know what the settings are going to be (unless we don't expose the controls in code).

Reply:
Yes, this is a real problem, I've talked with @kevingosse about this, and maybe the filtering at this level doesn't make sense. We can have similar issues with different AppDomains as well. So, because there's a lot of unknown and possible pitfalls and this is an optimisation only in those case when a user disables an integration I prefer to close this PR, and move the ADO.NET integrations name renaming and the runtime assembly name calculation to other PR.